src/common/s3util.py

```diff
@@ -144,13 +144,35 @@ def download_object(self, key, local_file_path):
     # get contents info from s3 folder
     def get_contents(self, prefix):
         contents = []
+        if not prefix.endswith('/'):
+            prefix += '/'
         try:
             for obj in self.bucket.objects.filter(Prefix=prefix):
                 # key end with ".tsv" or ".txt"
                 if obj.key[-4:] == ".tsv" or obj.key[-4:] == ".txt":
                     contents.append(obj.key)        
-        except Exception as e:
-            self.log.error("Failed to retrieve metadata file info")
+        except Exception:
+            self.log.error("Failed to retrieve child metadata files.")
+        finally:
+            return contents
+        
+    def get_contents_in_current_folder(self, prefix):
+        contents = []
+        if not prefix.endswith('/'):
+            prefix += '/'
+        try:
+            response = self.client.list_objects_v2(
```

Comment:
When there are more than 1000 files in a folder, list_objects_v2 will only return first 1000. We need to handle that situation by checking a flag returned by list_objects_v2.